WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -321,6 +336,7 @@ final class EditableOrderViewModel: ObservableObject {
     func createProductRowViewModel(for item: OrderItem, canChangeQuantity: Bool) -> ProductRowViewModel? {
         guard item.quantity > 0, // Don't render any item with `.zero` quantity.
               let product = allProducts.first(where: { $0.productID == item.productID }) else {
+            selectedProducts.removeAll(where: { $0?.productID == item.productID })
```

Comment:
I am wondering if this is the right place for doing this. After all, on this function we just want to create a `ProductRowViewModel` instance and return it, removing that item from the selected products is a side effect that is not expected from the function name.

Reply:
There seems to be better places indeed! I moved the method to `removeItemFromOrder(_:)` instead,  here: https://github.com/woocommerce/woocommerce-ios/pull/8943/commits/1e4d574bb5b4bd58be4a3f0c09c5c6398566c222 , as is where we specifically deal with synching, updating, and tracking when an item is removed from an Order.